pallets/pallet-bank/src/lib.rs

```diff
@@ -223,6 +241,8 @@ pub mod module {
         fn integrity_test() {
             // Check if the minimum deposit is greater than or equal to the existential deposit
             assert!(T::MinimumAmount::get() >= T::ExistentialDeposit::get());
+            assert!(T::InterestPayoutPeriod::get() <= T::StakePeriod::get());
+            assert!(T::InterestPayoutPeriod::get() <= T::RedeemPeriod::get());
```

Comment:
please add a check to ensure total blocks per year is > 0 (otherwise you will be dividing by 0)